### DISCUSSION TITLE
Website crashes when peaks array has large data

### DISCUSSION THREAD

**Discussion by fantasticoder** [katspaugh/wavesurfer.js#2917] [Q&A]
I am trying to load waveform using around 800,000 length of peaks array, but when loading, website crashes.
I think it's because peaks array length is too long.
Is there any way to render waveform properly without crashing website?

**Comment by katspaugh** [comment]
How are you loading them?

I've just tried with this code and it worked fine:
```
import WaveSurfer from 'https://unpkg.com/wavesurfer.js@beta'

const peaks = []
for (let i = 0; i < 800000; i++) {
  peaks.push(Math.sqrt(i))
}

WaveSurfer.create({
  container: document.body,
  waveColor: 'rgb(200, 0, 200)',
  progressColor: 'rgb(100, 0, 100)',
  url: '/examples/audio/audio.wav',
  peaks,
})
```

**Comment by fantasticoder** [comment]
can you try around 900,000 length?

**Reply by katspaugh** [reply]
It works fine even with 10 000 000 for me. You can try yourself on https://wavesurfer-js.org/examples/, just paste the code there.

**Comment by fantasticoder** [comment]
I think I am not suing the latest version.
I will try with the latest version.
Thanks @katspaugh

**Reply by katspaugh** [reply]
Phew, then we're not in trouble. :D
Cheers!
